src/irmin-pack/unix/ext.ml

```diff
@@ -351,6 +354,30 @@ module Maker (Config : Conf.S) = struct
                       Pack_key.v_direct ~offset ~length ~hash:entry.hash
                     in
                     Some key)
+
+          let create_one_commit_store t commit_key path =
+            let () = Io.mkdir path |> Errs.raise_if_error in
+            let* _launched =
+              start_exn ~use_auto_finalisation:false ~path_new_files:path t
+                commit_key
+            in
+            let generation = File_manager.generation t.fm + 1 in
+            let* () =
+              match t.running_gc with
+              | None -> assert false
+              | Some { gc; _ } -> Gc.finalise_without_swap gc
+            in
+            let () =
+              File_manager.create_one_commit_store t.config ~generation
+                ~new_store_root:path commit_key
+              |> Errs.raise_if_error
+            in
+            let branch_path = Irmin_pack.Layout.V4.branch ~root:path in
+            let* branch_store =
+              Branch.v ~fresh:true ~readonly:false branch_path
+            in
+            let* () = Branch.close branch_store in
```

Comment:
I wish this lived in `File_manager`'s part of this procedure so that all the additional file creation is in one place. Could be done if you passed the `Branch` module but I'm not sure how you feel about doing that.

Reply:
Ideally, it seems like file manager would have this responsibility. Why would we need to bump the control file version? It's not changing the actual files on disk, just which module is responsible for their lifecycle. Regardless, I agree it would be better placed in a separate PR.

Reply:
I think what @Ngoguey42 meant is that a proper integration of the branch store in the file manager means also an integration of the branch store with the control file. For now the branch store is using io_legacy which means it has a header in the file and does not need the control file. 

We could only "half" integrate the branch store in the file manager, by moving the code from ext to file manager, but I think even that is a bit of work and outside the scope of this PR. I opened an issue to keep track of this https://github.com/mirage/irmin/issues/2134.

Reply:
That makes sense to me. Thanks @icristescu!